enumerables.rb

```diff
@@ -84,8 +123,59 @@ def my_inject(acumulator = nil, operator = nil)
   def multiply_els
     my_inject(:*)
   end
+
+  def my_map(oper = nil)
+    res = []
+    return enum_for(__callee__) if !block_given? && oper.nil?
+
+    if block_given?
+      my_each { |i| res << yield(i) }
+    else
+      my_each { |i| res << oper.call(i) }
+    end
+    res
+  end
 end
```

Comment:
my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
```ruby
my_proc = Proc.new { |num| num > 10 }
[11, 2, 3, 15].my_map(my_proc) {|num| num < 10 } # should return [true, false, false, true]
```
Based on your condition you are checking if block_given? and it is finding block all the time based on the example that I have given. So you should add another clause that is and !oper. So the condition will become
```ruby
if block_given? and !oper
```

Reply:
Thanks we updated it